### PR TITLE
docs(api-gateway-usage-plan-key): correct example

### LINKED ISSUE #29913: [Docs]: Example usage on aws_api_gateway_usage_plan_key is incorrect
### Documentation Link

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_usage_plan_key#example-usage

### Description

See https://stackoverflow.com/a/72824794/2558542 

Currently it references `aws_api_gateway_deployment` when linking the `api_stage` but it should be `aws_api_gateway_stage`

### References

_No response_

### Would you like to implement a fix?

Yes

### PR DESCRIPTION
**Pull request by kiran94**
### Description

The current documentation provides example usage which includes: 

```terraform
resource "aws_api_gateway_usage_plan" "myusageplan" {
  name = "my_usage_plan"

  api_stages {
    api_id = aws_api_gateway_rest_api.test.id
    stage  = aws_api_gateway_deployment.foo.stage_name
  }
}
```

The `stage` field here references `aws_api_gateway_deployment`. If you look at the attributes available for that resource [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_deployment#attributes-reference) you can see it does not have `stage_name`. 

Instead, this should be referencing `aws_api_gateway_stage`. 

### Relations

Closes #29913 

### References

- [StackOverflow](https://stackoverflow.com/a/72824794/2558542)
- [`api_gateway_usage_plan_key` Documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_usage_plan_key#example-usage)
- [`aws_api_gateway_stage`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_stage#attributes-reference) Documentation

### Output from Acceptance Testing

NA

### CODE REVIEW

File: website/docs/r/api_gateway_stage.html.markdown [hunk 1]
```diff
@@ -142,6 +142,7 @@ In addition to all arguments above, the following attributes are exported:
   e.g., `arn:aws:execute-api:eu-west-2:123456789012:z4675bid1j/prod`
 * `tags_all` - Map of tags assigned to the resource, including those inherited from the provider [`default_tags` configuration block](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#default_tags-configuration-block).
 * `web_acl_arn` - ARN of the WebAcl associated with the Stage.
+* `stage_name` - Name of the stage
```
Comment (breathingdust) [inline website/docs/r/api_gateway_stage.html.markdown:145]:
```suggestion
```

I don't believe this change is necessary as its included as an argument: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_stage#stage_name

**Review by breathingdust — CHANGES_REQUESTED** [review]
One small change to make, this correction does seem inline with best practice as noted in the warning on the `deployment` resource:

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_deployment

After that we can merge this! Thanks for the contribution @kiran94! 🚀

File: website/docs/r/api_gateway_stage.html.markdown (hunk 1 above)
Comment (kiran94) [reply website/docs/r/api_gateway_stage.html.markdown]: Updated :smile:

**Review by justinretzolk — APPROVED** [review]
Looks good to me after the most recent change! Thank you again for your contribution! 🚀

**Pull request merged**